api/health/service_test.go

```diff
@@ -112,3 +114,130 @@ func TestServiceResponses(t *testing.T) {
 		require.True(reply.Healthy)
 	}
 }
+
+func TestServiceTagResponse(t *testing.T) {
+	require := require.New(t)
+
+	check := CheckerFunc(func(context.Context) (interface{}, error) {
+		return "", nil
+	})
+
+	subnetID1 := ids.GenerateTestID()
+	subnetID2 := ids.GenerateTestID()
+
+	// test cases
+	type testMethods struct {
+		name     string
+		register func(Health, string, Checker, ...string) error
+		check    func(*Service, *http.Request, *APIArgs, *APIReply) error
+		await    func(*testing.T, Reporter, bool)
+	}
+
+	tests := []testMethods{
+		{
+			name: "Readiness",
+			register: func(h Health, s1 string, c Checker, s2 ...string) error {
+				return h.RegisterReadinessCheck(s1, c, s2...)
+			},
+			check: func(s *Service, req *http.Request, a1 *APIArgs, a2 *APIReply) error {
+				return s.Readiness(req, a1, a2)
+			},
+			await: awaitReadiness,
+		},
+		{
+			name: "Health",
+			register: func(h Health, s1 string, c Checker, s2 ...string) error {
+				return h.RegisterHealthCheck(s1, c, s2...)
+			},
+			check: func(s *Service, r *http.Request, a1 *APIArgs, a2 *APIReply) error {
+				return s.Health(r, a1, a2)
+			},
+			await: awaitHealthy,
+		},
+		{
+			name: "Liveness",
+			register: func(h Health, s1 string, c Checker, s2 ...string) error {
+				return h.RegisterLivenessCheck(s1, c, s2...)
+			},
+			check: func(s *Service, r *http.Request, a1 *APIArgs, a2 *APIReply) error {
+				return s.Liveness(r, a1, a2)
+			},
+			await: awaitLiveness,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			h, err := New(logging.NoLog{}, prometheus.NewRegistry())
```

Comment:
So that if there's a failure, it reports the subtest that failed
```suggestion
		t.Run(test.name, func(t *testing.T) {
		        require := require.New(t)
			h, err := New(logging.NoLog{}, prometheus.NewRegistry())
```

Reply:
ah yes, good catch.